### PR TITLE
bme280: Support sensor shell correctly

### PR DESCRIPTION
**Pull request by ydamigos**
Update bme280 sensor driver to support
sensor shell correctly.

Fixes issue #74927

### CODE REVIEW

**Review by laurenmurphyx64 — DISMISSED** [review]
Looks / runs good for me, thanks for fixing this!

File: drivers/sensor/bosch/bme280/bme280.h [hunk 1]
```diff
@@ -206,6 +206,14 @@ struct bme280_decoder_header {
 
 struct bme280_encoded_data {
 	struct bme280_decoder_header header;
+	struct {
+		/** Set if `accel_xyz` has data */
```
Comment (ubieda) [inline drivers/sensor/bosch/bme280/bme280.h:210]: Nit: 
```suggestion
		/** Set if `temp` has data */
```

**Review by ubieda — COMMENTED** [review]
Looks good, but the commit message is misleading as the fix has nothing to do with Shell, but rather the driver itself.

Can you update it to better reflect the fix?

File: drivers/sensor/bosch/bme280/bme280.h (hunk 1 above)
Comment (ydamigos) [reply drivers/sensor/bosch/bme280/bme280.h]: Fixed

**Comment by ydamigos** [comment]
> Looks good, but the commit message is misleading as the fix has nothing to do with Shell, but rather the driver itself.
> 
> Can you update it to better reflect the fix?

I updated the commit message.

**Review by ubieda — APPROVED** [review]
Thanks!

File: drivers/sensor/bosch/bme280/bme280.h [hunk 2]
```diff
@@ -206,6 +206,14 @@ struct bme280_decoder_header {
 
 struct bme280_encoded_data {
 	struct bme280_decoder_header header;
+	struct {
```
Comment (teburd) [inline drivers/sensor/bosch/bme280/bme280.h:209]: This seems like it should be in the decoder header

**Review by teburd — APPROVED** [review]
Clearly fixes a bug, one nit comment (not a blocker)

File: drivers/sensor/bosch/bme280/bme280.h (hunk 2 above)
Comment (ydamigos) [reply drivers/sensor/bosch/bme280/bme280.h:209]: I used BMA4xx driver as a "guide" and tried to be consistent.

```
struct bma4xx_encoded_data {
	struct bma4xx_decoder_header header;
	struct {
		/** Set if `accel_xyz` has data */
		uint8_t has_accel: 1;
		/** Set if `temp` has data */
		uint8_t has_temp: 1;
		uint8_t reserved: 6;
	} __attribute__((__packed__));
	int16_t accel_xyz[3];
#ifdef CONFIG_BMA4XX_TEMPERATURE
	int8_t temp;
#endif /* CONFIG_BMA4XX_TEMPERATURE */
};
```

Comment (teburd) [reply drivers/sensor/bosch/bme280/bme280.h:209]: Fair enough!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**